crates/router/src/connector/bambora/transformers.rs

```diff
@@ -0,0 +1,453 @@
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        data: (
+            types::ResponseRouterData<F, BamboraPaymentsResponse, T, types::PaymentsResponseData>,
+            PaymentFlow,
+        ),
+    ) -> Result<Self, Self::Error> {
+        let flow = data.1;
+        let item = data.0;
+        let pg_response = item.response;
+        Ok(Self {
+            status: match pg_response.approved.as_str() {
+                "0" => match flow {
+                    PaymentFlow::Authorize => enums::AttemptStatus::AuthorizationFailed,
+                    PaymentFlow::Capture => enums::AttemptStatus::Failure,
+                    PaymentFlow::Void => enums::AttemptStatus::VoidFailed,
+                },
+                "1" => match flow {
+                    PaymentFlow::Authorize => enums::AttemptStatus::Authorized,
+                    PaymentFlow::Capture => enums::AttemptStatus::Charged,
+                    PaymentFlow::Void => enums::AttemptStatus::Voided,
+                },
+                &_ => Err(errors::ConnectorError::ResponseDeserializationFailed)?,
+            },
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(pg_response.id.to_string()),
+                redirection_data: None,
+                mandate_reference: None,
+                connector_metadata: None,
+            }),
+            ..item.data
+        })
+    }
+}
+
+fn str_or_i32<'de, D>(deserializer: D) -> Result<String, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    #[derive(Deserialize)]
+    #[serde(untagged)]
+    enum StrOrI32 {
+        Str(String),
+        I32(i32),
+    }
+
+    let value = StrOrI32::deserialize(deserializer)?;
+    let res = match value {
+        StrOrI32::Str(v) => v,
+        StrOrI32::I32(v) => v.to_string(),
+    };
+    Ok(res)
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct BamboraPaymentsResponse {
+    #[serde(deserialize_with = "str_or_i32")]
+    id: String,
+    authorizing_merchant_id: i32,
+    #[serde(deserialize_with = "str_or_i32")]
+    approved: String,
+    #[serde(deserialize_with = "str_or_i32")]
+    message_id: String,
+    message: String,
+    auth_code: String,
+    created: String,
+    amount: f32,
+    order_number: String,
+    #[serde(rename = "type")]
+    payment_type: String,
+    comments: Option<String>,
+    batch_number: Option<String>,
+    total_refunds: Option<f32>,
+    total_completions: Option<f32>,
+    payment_method: String,
+    card: CardData,
+    billing: Option<AddressData>,
+    shipping: Option<AddressData>,
+    custom: CustomData,
+    adjusted_by: Option<Vec<AdjustedBy>>,
+    links: Vec<Links>,
+    risk_score: Option<f32>,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct CardData {
+    name: Option<String>,
```

Comment:
Make all PII fields as Secrets, Why we have two types for cards

Reply:
cards data from response is already masked/4 digits, so masking again is not needed